### PR TITLE
vkd3d: Introduce VKD3D_FILTER_DEVICE_NAME.

### PR DESCRIPTION
**Pull request by DadSchoorse**
The same as `DXVK_FILTER_DEVICE_NAME`.

### CODE REVIEW

File: libs/vkd3d/device.c [hunk 1]
```diff
@@ -1640,6 +1644,12 @@ static HRESULT vkd3d_select_physical_device(struct vkd3d_instance *instance,
         if (i == device_index)
             device = physical_devices[i];
 
+        if (!strstr(device_properties.deviceName, filter))
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:1647]: Is it well defined how strstr works with empty string? Seems cleaner to check `if (filter && strstr())`.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Small nit.

File: libs/vkd3d/device.c (hunk 1 above)
Comment (DadSchoorse) [reply libs/vkd3d/device.c]: Yes, it's well defined (it will return the first argument). But yes, `if (filter && !strstr()).` is probably cleaner.

**Review by HansKristian-Work — APPROVED** [review]
LGTM, thanks.

**Pull request merged**